core/application/bloodDonationWorkflow/BloodDonationService.ts

```diff
@@ -0,0 +1,58 @@
+import { GENERIC_CODES } from '../../../commons/libs/constants/GenericCodes'
+import BloodDonationOperationError from './BloodDonationOperationError'
+import { DonationDTO } from '../../../commons/dto/DonationDTO'
+import { generateUniqueID } from '../utils/idGenerator'
+import Repository from '../technicalImpl/policies/repositories/Repository'
+import { generateGeohash } from '../utils/geohash'
+import { validateInputWithRules } from '../utils/validator'
+import { BloodDonationAttributes, validationRules, UpdateBloodDonationAttributes } from './Types'
+
+export class BloodDonationService {
+  async createBloodDonation(donationAttributes: BloodDonationAttributes, bloodDonationRepository: Repository<DonationDTO>): Promise<string> {
+    try {
+      const validationResponse = validateInputWithRules({ bloodQuantity: donationAttributes.bloodQuantity, donationDateTime: donationAttributes.donationDateTime }, validationRules)
+      if (validationResponse !== null) {
+        return validationResponse
+      }
+      await bloodDonationRepository.create({
+        id: generateUniqueID(),
+        ...donationAttributes,
+        status: 'accepted',
+        geohash: generateGeohash(donationAttributes.latitude, donationAttributes.longitude),
+        donationDateTime: new Date(donationAttributes.donationDateTime).toISOString()
+      })
+      return 'We have accepted you request we will let you know when we will found donar.'
```

Comment:
incorrect grammer. 
We have accepted you request and we will let you know when we find a donor.

maybe send back a request id as well

Reply:
Vai, This is my understanding, but I might be completely wrong.

After we create a blood request post and it succeeds, we might be directed to a page displaying all the posts. At that point, the new post we just created will also be visible, along with its request ID.